src/qibolab/platforms/native.py

```diff
@@ -0,0 +1,188 @@
+
+        Args:
+            name (str): Name of the gate the sequence is applying.
+            qubits (list): List of :class:`qibolab.platforms.abstract.Qubit` object for all
+                qubits in the platform. All qubits are required because the sequence may be
+                acting on qubits that the implemented gate is not targeting.
+            sequence (dict): Dictionary describing the sequence as provided in the runcard.
+        """
+        pulses = []
+
+        # If sequence contains only one pulse dictionary, convert it into a list that can be iterated below
+        if isinstance(sequence, dict):
+            sequence = [sequence]
+
+        for i, pulse in enumerate(sequence):
+            pulse_type = pulse["type"]
+            kwargs = dict(pulse)
+            qubit = qubits[kwargs.pop("qubit")]
+            if pulse_type == "virtual_z":
+                phase = kwargs["phase"]
+                pulses.append(VirtualPulse(phase, qubit))
+            elif pulse_type == "qf":
+                name = f"{name}{i}"
+                pulses.append(NativeFluxPulse.from_dict(name, **kwargs, qubit=qubit))
+            elif pulse_type == "qd":
+                name = f"{name}{i}"
+                pulses.append(NativePulse.from_dict(name, **kwargs, qubit=qubit))
+            else:
+                raise_error(TypeError, f"Incorrect type {pulse_type} was given in NativeSequence.")
+
+        return cls(name, pulses)
+
+    def sequence(self, start=0):
+        """Creates a :class:`qibolab.pulses.PulseSequence` object implementing the sequence."""
+        sequence = PulseSequence()
+        virtual_z_phases = defaultdict(int)
+
+        for pulse in self.pulses:
+            if isinstance(pulse, NativePulse):
+                sequence.add(pulse.pulse(start=start))
+            else:
+                virtual_z_phases[pulse.qubit.name] += pulse.phase
+
+        # TODO: Maybe ``virtual_z_phases`` should be an attribute of ``PulseSequence``
+        return sequence, virtual_z_phases
+
+
+@dataclass
+class NativeSingleQubitGates:
+    """Container with the native single qubit gates acting on a specific qubit."""
+
+    MZ: NativePulse
+    RX: NativePulse
+    RX90: NativePulse
+
+    @classmethod
+    def from_dict(cls, qubit, native_gates):
+        """Parse native gates of the qubit from the runcard.
+
+        Args:
+            qubit (:class:`qibolab.platforms.abstract.Qubit`): Qubit object that the
+                native gates are acting on.
+            native_gates (dict): Dictionary with native gate pulse parameters as loaded
+                from the runcard.
+        """
+        mz = NativePulse.from_dict("MZ", **native_gates["MZ"], qubit=qubit)
+        rx = NativePulse.from_dict("RX", **native_gates["RX"], qubit=qubit)
+        rx90 = NativePulse.from_dict("RX90", **native_gates["RX"], qubit=qubit)
+        rx90.amplitude /= 2.0
+        return cls(mz, rx, rx90)
```

Comment:
Here it's me being sick, but that's some repeated code:
```suggestion
        pulses = {n: NativePulse.from_dict(n, **native_gates[n], qubit=qubit) for n in ["MZ", "RX", "RX90"]}
        pulses["rx90"].amplitude /= 2.0
        return cls(**pulses)
```